docs/core/extensions/httpclient-factory.md

```diff
@@ -235,6 +243,41 @@ There are several additional configuration options for controlling the `IHttpCli
 | <xref:Microsoft.Extensions.DependencyInjection.HttpClientBuilderExtensions.RedactLoggedHeaders%2A> | Sets the collection of HTTP header names for which values should be redacted before logging. |
 | <xref:Microsoft.Extensions.DependencyInjection.HttpClientBuilderExtensions.SetHandlerLifetime%2A> | Sets the length of time that a `HttpMessageHandler` instance can be reused. Each named client can have its own configured handler lifetime value. |
 
+## Using HttpClientFactory together with SocketsHttpHandler
+
+Since .NET Core 2.1, `SocketsHttpHandler` implementation of `HttpMessageHandler` was added, that allows configuring `PooledConnectionLifetime`. This setting is used to ensure that the handler will react to DNS changes, so using `SocketsHttpHandler` is considered to be an alternative to using `IHttpClientFactory`, see [Guidelines for using HTTP clients](../../fundamentals/networking/http/httpclient-guidelines.md).
+
+However, `SocketsHttpHandler` and `IHttpClientFactory` can be used together, to benefit from configurability on both low level (e.g. using `LocalCertificateSelectionCallback` for dynamic certificate selection) and high level (e.g. leveraging DI integration and several client configurations).
+
+To do that, you need to
+1. Specify `SocketsHttpHandler` as `PrimaryHandler` and set up it's `PooledConnectionLifetime` (e.g. to a value that was previously in `HandlerLifetime`)
+2. As `SocketsHttpHandler` will handle connection pooling and recycling, then `IHttpClientFactory`-level handler recycling is not needed anymore, so it could be disabled by setting `HandlerLifetime` to `Timeout.InfiniteTimeSpan`.
+
+```csharp
+services.AddHttpClient(name)
+    .ConfigurePrimaryHttpMessageHandler(() =>
+    {
+        return new SocketsHttpHandler()
+        {
+            PooledConnectionLifetime = TimeSpan.FromMinutes(handlerLifetimeInMinutes)
+        };
+    })
+    .SetHandlerLifetime(Timeout.InfiniteTimeSpan); // Disable rotation, as it is handled by PooledConnectionLifetime
+```
+
+## Using Typed clients in singleton services
+
+When Named client approach is used, `IHttpClientFactory` is injected into services, and `HttpClient`s are created by calling <xref:System.Net.Http.IHttpClientFactory.CreateClient%2A> every time an `HttpClient` is needed.
+
+However, with Typed client approach, Typed clients are transient objects usually injected into services. That may cause a problem, because a Typed client can be injected into a singleton service.
```

Comment:
```suggestion
However, with the _typed client_ approach, typed clients are transient objects usually injected into services. That may cause a problem because a typed client can be injected into a singleton service.
```